crates/egui-winit/src/lib.rs

```diff
@@ -1310,11 +1327,17 @@ fn process_viewport_command(
         ViewportCommand::InnerSize(size) => {
             let width_px = pixels_per_point * size.x.max(1.0);
             let height_px = pixels_per_point * size.y.max(1.0);
-            if window
-                .request_inner_size(PhysicalSize::new(width_px, height_px))
-                .is_some()
+            if let Some(_inner_size) =
+                window.request_inner_size(PhysicalSize::new(width_px, height_px))
             {
-                log::debug!("ViewportCommand::InnerSize ignored by winit");
+                // ex) linux
+                log::info!("ViewportCommand::InnerSize, not will be delivered later with the [WindowEvent::Resized]");
+
+                info.inner_rect = math_inner_rect(window, Some(pixels_per_point));
+                info.outer_rect = math_outer_rect(window, Some(pixels_per_point));
```

Comment:
From the docs of `request_inner_size`:

> On platforms where the size is entirely controlled by the user the applied size will be returned immediately, resize event in such case may not be generated.
> On platforms where resizing is disallowed by the windowing system, the current inner size is returned immidiatelly, and the user one is ignored.

Your code seems to assume we're in the first case, i.e. that the size was applied, but we won't be getting any resized events.

We should check if we're not in the second case. If the returned size is the same as the one we passed to `request_inner_size`, then we're in the first case. If not, we're in the second (and should just log a warning or something).